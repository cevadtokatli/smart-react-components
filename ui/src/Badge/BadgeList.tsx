import extractElementProps from '@smart-react-components/core/element-props'
import intrinsicStyledCoreProps, { IntrinsicStyledCoreProps } from '@smart-react-components/core/element-props/intrinsic-styled-core-props'
import { JSXElementProps, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import BadgeListElement from '../components/Badge/BadgeListElement'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  IntrinsicStyledCoreProps {
  children: JSX.Element[]
  elementProps?: JSXElementProps
  hasSpace?: boolean
  isFixedSize?: boolean
  isOutline?: boolean
  isSoft?: boolean
  palette?: PaletteProp
  shape?: ShapeProp
}

const BadgeList: React.FC<Props> = props => (
  <BadgeListElement
    badgeListSize={props.size ?? 'medium'}
    badgeListSizeSm={props.sizeSm}
    badgeListSizeMd={props.sizeMd}
    badgeListSizeLg={props.sizeLg}
    badgeListSizeXl={props.sizeXl}
    hasSpace={props.hasSpace ?? true}
    {...extractElementProps(props, [intrinsicStyledCoreProps])}
    {...props.elementProps}
  >
    { props.children.map((item, index) => React.cloneElement(item, {
      key: item.key ?? index,
      hasSpace: props.hasSpace ?? item.props.hasSpace ?? true,
      isFixedSize: props.isFixedSize ?? item.props.isFixedSize,
      isOutline: props.isOutline ?? item.props.isOutline,
      isSoft: props.isSoft ?? item.props.isSoft,
      palette: props.palette ?? item.props.palette,
      shape: props.shape ?? item.props.shape,
      size: props.size ?? item.props.size,
      sizeSm: props.sizeSm ?? item.props.sizeSm,
      sizeMd: props.sizeMd ?? item.props.sizeMd,
      sizeLg: props.sizeLg ?? item.props.sizeLg,
      sizeXl: props.sizeXl ?? item.props.sizeXl,
    })) }
  </BadgeListElement>
)

export default BadgeList
