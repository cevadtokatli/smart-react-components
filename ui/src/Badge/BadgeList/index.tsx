import extractElementProps from '@smart-react-components/core/element-props'
import intrinsicStyledProps, { IntrinsicStyledProps } from '@smart-react-components/core/element-props/intrinsic-styled-props'
import { JSXElementProps, PaletteProp, Partial, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import BadgeListElement from './BadgeListElement'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  IntrinsicStyledProps {
  children: JSX.Element[]
  elementProps?: JSXElementProps
  hasSpace?: boolean
  isFixedSize?: boolean
  isOutline?: boolean
  isSoft?: boolean
  palette?: PaletteProp
  shape?: ShapeProp
}

const BadgeList: React.FC<Props> = props => (
  <BadgeListElement
    badgeListSize={props.size}
    badgeListSizeSm={props.sizeSm}
    badgeListSizeMd={props.sizeMd}
    badgeListSizeLg={props.sizeLg}
    badgeListSizeXl={props.sizeXl}
    hasSpace={props.hasSpace}
    {...extractElementProps(props, [intrinsicStyledProps])}
    {...props.elementProps}
  >
    { props.children.map((item, index) => React.cloneElement(item, {
      key: item.key ?? index,
      hasSpace: props.hasSpace,
      isFixedSizze: props.isFixedSize,
      isOutline: props.isOutline,
      isSoft: props.isSoft,
      palette: props.palette,
      shape: props.shape,
      size: props.size,
      sizeSm: props.sizeSm,
      sizeMd: props.sizeMd,
      sizeLg: props.sizeLg,
      sizeXl: props.sizeXl,
      ...item.props,
    })) }
  </BadgeListElement>
)

BadgeList.defaultProps = {
  hasSpace: true,
  size: 'medium',
}

export default BadgeList
