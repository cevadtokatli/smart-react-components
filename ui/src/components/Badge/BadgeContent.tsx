import styled from 'styled-components'

export default styled.div.attrs({
  className: 'src-badge-content',
})`
  border: solid 1px;
  flex: 0 1 auto;
`
