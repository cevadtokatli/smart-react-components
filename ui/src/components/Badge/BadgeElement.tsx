import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { toCSSValue } from '../../util/css'

interface Props extends
  StyledProps,
  ResponsiveProp<'badgeSize', SizeProp> {
  hasIconLeft: boolean
  hasIconRight: boolean
  hasSpace: boolean
  isClickable: boolean
  isFixedSize: boolean
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  shape: ShapeProp
}

export default styled(Div).attrs<Props>(({ hasIconLeft, hasIconRight, hasSpace, isFixedSize }) => ({
  getBadgeSize: (v, t) => `
    ${hasSpace
      ? `
        margin: ${t.$.size.badge[v].margin.y} ${t.$.size.badge[v].margin.x};    
      `
      : ''
    }

    .src-badge-content {
      font-size: ${t.$.size.badge[v].fontSize};

      ${!isFixedSize
        ? `
          padding: ${t.$.size.badge[v].padding.y} ${t.$.size.badge[v].padding.x};

          ${hasIconLeft
            ? `
              padding-left: ${toCSSValue(t.$.size.badge[v].padding.x)(v => v / 2)};
            `
            : ''
          }

          ${hasIconRight
            ? `
              padding-right: ${toCSSValue(t.$.size.badge[v].padding.x)(v => v / 2)};  
            `
            : ''
          }
        `
        : `
          height: ${t.$.size.badge[v].fixedSize};
          width: ${t.$.size.badge[v].fixedSize};
        `
      }
    }

    .src-badge-icon {
      padding: 0 ${toCSSValue(t.$.size.badge[v].padding.x)(value => value / 2)};
    }

    svg {
      height: ${t.$.size.badge[v].iconSize};
      width: ${t.$.size.badge[v].iconSize};
    }

    i {
      font-size: ${t.$.size.badge[v].iconSize};
    }
  `,
}))<Props>(({ theme, hasIconLeft, hasIconRight, isClickable, isFixedSize, isOutline, isSoft, palette, shape }: Props) => `
  display: inline-flex;
  font-family: ${theme.$.fontFamily.badge};
  word-break: break-word;
  word-wrap: break-word;

  .src-badge-content {
    border-radius: ${(isFixedSize && shape === 'rounded') ? '100%' : theme.$.radius.badge[shape]};
  }

  ${isSoft
    ? `
      font-weight: ${theme.$.fontWeight.bold};
    `
    : ''
  }

  ${!isOutline
    ? `  
      .src-badge-content {
        ${!isSoft
          ? `
            background: ${theme.$.palette[palette].badge?.background ?? theme.$.palette[palette].main};
            border-color: ${theme.$.palette[palette].badge?.border ?? theme.$.palette[palette].main};
            color: ${theme.$.palette[palette].badge?.font ?? theme.$.palette[palette].font};
            fill: ${theme.$.palette[palette].badge?.font ?? theme.$.palette[palette].font};
          `
          : `
            background: ${theme.$.palette[palette].soft};
            border-color: ${theme.$.palette[palette].soft};
            color: ${theme.$.palette[palette].softFont};
            fill: ${theme.$.palette[palette].softFont};
          `
        }
      }

      ${isClickable
        ? `
          cursor: ${theme.$.cursor.badgeClickable};

          &:active {
            ${!isSoft
              ? `
                .src-badge-content {
                  background: ${theme.$.palette[palette].dynamicer};
                  border-color: ${theme.$.palette[palette].dynamicer};
                }

                .src-badge-icon {
                  background: ${theme.$.palette[palette].dynamicest};
                  border-color: ${theme.$.palette[palette].dynamicest};
                }
              `
              : `
                .src-badge-content {
                  background: ${theme.$.palette[palette].softDynamicer};
                  border-color: ${theme.$.palette[palette].softDynamicer};
                }

                .src-badge-icon {
                  background: ${theme.$.palette[palette].softDynamicest};
                  border-color: ${theme.$.palette[palette].softDynamicest};
                }
              `
            }
          }
        `
        : ''
      }
    `
    : `
      .src-badge-content {
        ${!isSoft
          ? `
            border-color: ${theme.$.palette[palette].badge?.outline?.border ?? theme.$.palette[palette].main};
            color: ${theme.$.palette[palette].badge?.outline?.font ?? theme.$.palette[palette].main};
            fill: ${theme.$.palette[palette].badge?.outline?.font ?? theme.$.palette[palette].main};
          `
          : `
            border-color: ${theme.$.palette[palette].soft};
            color: ${theme.$.palette[palette].soft};
            fill: ${theme.$.palette[palette].soft};
          `
        }
      }

      ${isClickable
        ? `
          &:active {
            ${!isSoft
              ? `
               .src-badge-content {
                  background: ${theme.$.palette[palette].main};
                  color: ${theme.$.palette[palette].font};
                  fill: ${theme.$.palette[palette].font};
                }

                .src-badge-icon {
                  background: ${theme.$.palette[palette].dynamicer};
                  color: ${theme.$.palette[palette].font};
                  fill: ${theme.$.palette[palette].font}; 
                }
              `
              : `
               .src-badge-content {
                  background: ${theme.$.palette[palette].soft};
                  color: ${theme.$.palette[palette].softFont};
                  fill: ${theme.$.palette[palette].softFont};
                }

                .src-badge-icon {
                  background: ${theme.$.palette[palette].softDynamicer};
                  color: ${theme.$.palette[palette].softFont};
                  fill: ${theme.$.palette[palette].softFont};
                }
              `
            }
          }
        `
        : ''
      }
    `
  }

  .src-badge-content {
    ${isFixedSize
      ? `
        align-items: center;
        display: inline-flex;
        justify-content: center;
      `
      : ''
    }

    ${hasIconLeft
      ? `
        border-left: 0;
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;
      `
      : ''
    }
  
    ${hasIconRight
      ? `
        border-right: 0;
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      `
      : ''
    }
  }

  ${isClickable
    ? `
      cursor: default;
      user-select: none;
    `
    : ''
  }
`)
