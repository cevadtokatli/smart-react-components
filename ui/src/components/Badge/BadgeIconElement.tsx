import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ShapeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { OrderPosition } from '../../types'
import { getReverseOrderPosition } from '../../util/props'

interface Props extends StyledProps {
  iconPosition: OrderPosition
  isClickable: boolean
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  shape: ShapeProp
}

export default styled(Div)<Props>(({ theme, iconPosition, isClickable, isOutline, isSoft, palette, shape }: Props) => `
  align-items: center;
  align-self: stretch;
  border: solid 1px;
  border-${OrderPosition[getReverseOrderPosition(iconPosition)].toLowerCase()}: 0;
  border-color: ${!isSoft ? theme.$.palette[palette].dynamicer : theme.$.palette[palette].softDarker};  
  border-radius: ${theme.$.radius.badge[shape]};
  border-bottom-${OrderPosition[getReverseOrderPosition(iconPosition)].toLowerCase()}-radius: 0;
  border-top-${OrderPosition[getReverseOrderPosition(iconPosition)].toLowerCase()}-radius: 0;
  display: inline-flex;
  flex: 0 0 auto;
  justify-content: center;

  ${!isOutline
    ? `  
      ${!isSoft
        ? `
          background: ${theme.$.palette[palette].dynamicer};
          color: ${theme.$.palette[palette].font};
          fill: ${theme.$.palette[palette].font};
        `
        : `
          background: ${theme.$.palette[palette].softDarker};
          color: ${theme.$.palette[palette].softFont};
          fill: ${theme.$.palette[palette].softFont};       
        `
      }

      ${isClickable
        ? `
          cursor: ${theme.$.cursor.badgeIconClickable};

          &:active {
            ${!isSoft
              ? `
                background: ${theme.$.palette[palette].dynamicest};
                border-color: ${theme.$.palette[palette].dynamicest};  
              `
              : `
                background: ${theme.$.palette[palette].softDarkest};
                border-color: ${theme.$.palette[palette].softDarkest};
              `
            }
          }
        `
        : ''
      }
    `
    : `
     ${!isSoft
        ? `
          color: ${theme.$.palette[palette].main};
          fill: ${theme.$.palette[palette].main};
        `
        : `
          color: ${theme.$.palette[palette].soft};
          fill: ${theme.$.palette[palette].soft};       
        `
      }

      ${isClickable
        ? `
          &:active {
            ${!isSoft
              ? `
                background: ${theme.$.palette[palette].dynamicer};
                color: ${theme.$.palette[palette].font};
                fill: ${theme.$.palette[palette].font};
              `
              : `
                background: ${theme.$.palette[palette].softDarker};
                color: ${theme.$.palette[palette].softFont};
                fill: ${theme.$.palette[palette].softFont};
              `
            }
          }
        `
        : ''
      }
    `
  }
`)
