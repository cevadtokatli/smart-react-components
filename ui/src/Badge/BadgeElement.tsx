import Div from '@smart-react-components/core/Element/Div'
import { StyledProps } from '@smart-react-components/core/styled-props'
import { PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import styled from 'styled-components'
import { Position } from '../types'
import { toCSSValue } from '../util/css'
import BadgeIconElement from './BadgeIcon/BadgeIconElement'

export const Content = styled.div`
  border: solid 1px;
  flex: 0 1 auto;
`

interface Props extends
  StyledProps,
  ResponsiveProp<'badgeSize', SizeProp> {
  badgePosition: Position
  hasIconLeft: boolean
  hasIconRight: boolean
  hasSpace: boolean
  isAbsolute: boolean
  isClickable: boolean
  isFixedSize: boolean
  isOutline: boolean
  isSoft: boolean
  offset: number
  palette: PaletteProp
  shape: ShapeProp
}

export default styled(Div).attrs<Props>(({ hasIconLeft, hasIconRight, hasSpace, isFixedSize }) => ({
  getBadgeSize: (v, t) => `
    ${hasSpace
      ? `
        margin: ${t.$.size.badge[v].margin.y} ${t.$.size.badge[v].margin.x};    
      `
      : ''
    }

    ${Content} {
      font-size: ${t.$.size.badge[v].fontSize};

      ${!isFixedSize
        ? `
          padding: ${t.$.size.badge[v].padding.y} ${t.$.size.badge[v].padding.x};

          ${hasIconLeft
            ? `
              padding-left: ${toCSSValue(t.$.size.badge[v].padding.x)(v => v / 2)};
            `
            : ''
          }

          ${hasIconRight
            ? `
              padding-right: ${toCSSValue(t.$.size.badge[v].padding.x)(v => v / 2)};  
            `
            : ''
          }
        `
        : `
          height: ${t.$.size.badge[v].fixedSize};
          width: ${t.$.size.badge[v].fixedSize};
        `
      }
    }

    ${BadgeIconElement} {
      padding: 0 ${toCSSValue(t.$.size.badge[v].padding.x)(value => value / 2)};
    }

    svg {
      height: ${t.$.size.badge[v].iconSize};
      width: ${t.$.size.badge[v].iconSize};
    }

    i {
      font-size: ${t.$.size.badge[v].iconSize};
    }
  `,
}))<Props>(({ theme, badgePosition, hasIconLeft, hasIconRight, isAbsolute, isClickable, isFixedSize, isOutline, isSoft, offset, palette, shape }: Props) => `
  display: inline-flex;
  font-family: ${theme.$.fontFamily.badge};
  word-break: break-word;
  word-wrap: break-word;

  ${Content} {
    border-color: ${!isSoft ? theme.$.palette[palette].main : theme.$.palette[palette].soft};  
    border-radius: ${(isFixedSize && shape === 'rounded') ? '100%' : theme.$.radius.badge[shape]};
  }

  ${!isOutline
    ? `  
      ${Content} {
        ${!isSoft
          ? `
            background: ${theme.$.palette[palette].main};
            color: ${theme.$.palette[palette].font};
            fill: ${theme.$.palette[palette].font};
          `
          : `
            background: ${theme.$.palette[palette].soft};
            color: ${theme.$.palette[palette].softFont};
            fill: ${theme.$.palette[palette].softFont};
          `
        }
      }

      ${isClickable
        ? `
          &:active {
            ${!isSoft
              ? `
                ${Content} {
                  background: ${theme.$.palette[palette].dynamicer};
                  border-color: ${theme.$.palette[palette].dynamicer};
                }

                ${BadgeIconElement} {
                  background: ${theme.$.palette[palette].dynamicest};
                  border-color: ${theme.$.palette[palette].dynamicest};  
                }
              `
              : `
                ${Content} {
                  background: ${theme.$.palette[palette].softDarker};
                  border-color: ${theme.$.palette[palette].softDarker};
                }

                ${BadgeIconElement} {
                  background: ${theme.$.palette[palette].softDarkest};
                  border-color: ${theme.$.palette[palette].softDarkest};
                }
              `
            }
          }
        `
        : ''
      }
    `
    : `
      ${Content} {
        ${!isSoft
          ? `
            color: ${theme.$.palette[palette].main};
            fill: ${theme.$.palette[palette].main};
          `
          : `
            color: ${theme.$.palette[palette].soft};
            fill: ${theme.$.palette[palette].soft};
          `
        }
      }

      ${isClickable
        ? `
          &:active {
            ${!isSoft
              ? `
                ${Content} {
                  background: ${theme.$.palette[palette].main};
                  color: ${theme.$.palette[palette].font};
                  fill: ${theme.$.palette[palette].font};
                }

                ${BadgeIconElement} {
                  background: ${theme.$.palette[palette].dynamicer};
                  color: ${theme.$.palette[palette].font};
                  fill: ${theme.$.palette[palette].font}; 
                }
              `
              : `
                ${Content} {
                  background: ${theme.$.palette[palette].soft};
                  color: ${theme.$.palette[palette].softFont};
                  fill: ${theme.$.palette[palette].softFont};
                }

                ${BadgeIconElement} {
                  background: ${theme.$.palette[palette].softDarker};
                  color: ${theme.$.palette[palette].softFont};
                  fill: ${theme.$.palette[palette].softFont};
                }
              `
            }
          }
        `
        : ''
      }
    `
  }

  ${Content} {
    ${isFixedSize
      ? `
        align-items: center;
        display: inline-flex;
        justify-content: center;
      `
      : ''
    }

    ${hasIconLeft
      ? `
        border-left: 0;
        border-bottom-left-radius: 0;
        border-top-left-radius: 0;
      `
      : ''
    }
  
    ${hasIconRight
      ? `
        border-right: 0;
        border-bottom-right-radius: 0;
        border-top-right-radius: 0;
      `
      : ''
    }
  }

  ${isClickable
    ? `
      cursor: default;
      user-select: none;
    `
    : ''
  }

  ${isAbsolute
    ? `
      position: absolute;
      ${Position[badgePosition].toLowerCase()}: ${offset}px;
    `
    : ''
  }
`)
