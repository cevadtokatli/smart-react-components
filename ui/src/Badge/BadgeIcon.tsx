import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import mouseClickEvents from '@smart-react-components/core/element-props/mouse-click-events'
import { PaletteProp, ShapeProp } from '@smart-react-components/core/types'
import React from 'react'
import { OrderPosition } from '../types'
import BadgeIconElement from '../components/Badge/BadgeIconElement'

export interface Props extends ClickEvents {
  children: JSX.Element
}

interface PrivateProps {
  isOutline: boolean
  isSoft: boolean
  palette: PaletteProp
  position: OrderPosition
  shape: ShapeProp
}

const BadgeIcon: React.FC<Props> = (props: Props & PrivateProps) => (
  <BadgeIconElement
    {...extractElementProps(props, [clickEvents])}
    iconPosition={props.position}
    isClickable={!!mouseClickEvents.find(i => props[i])}
    isOutline={props.isOutline}
    isSoft={props.isSoft}
    palette={props.palette}
    shape={props.shape}
  >{props.children}</BadgeIconElement>
)

BadgeIcon.displayName = 'SRCBadgeIcon'

export default BadgeIcon
