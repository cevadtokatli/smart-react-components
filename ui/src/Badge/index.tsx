import extractElementProps from '@smart-react-components/core/element-props'
import clickEvents, { ClickEvents } from '@smart-react-components/core/element-props/click-events'
import intrinsicStyledProps, { IntrinsicStyledProps } from '@smart-react-components/core/element-props/intrinsic-styled-props'
import mouseClickEvents from '@smart-react-components/core/element-props/mouse-click-events'
import useChangeEffect from '@smart-react-components/core/hooks/useChangeEffect'
import { ContentElement, PaletteProp, ResponsiveProp, ShapeProp, SizeProp } from '@smart-react-components/core/types'
import React from 'react'
import { Position } from '../types'
import { extractIconsOutOfChildren } from '../util/props'
import BadgeElement, { Content } from './BadgeElement'

export interface Props extends
  Partial<ResponsiveProp<'size', SizeProp>>,
  ClickEvents,
  IntrinsicStyledProps {
  children: ContentElement | ContentElement[]
  hasSpace?: boolean
  isAbsolute?: boolean
  isFixedSize?: boolean
  isOutline?: boolean
  isSoft?: boolean
  palette?: PaletteProp
  position?: Position
  offset?: number
  shape?: ShapeProp
}

const Badge: React.FC<Props> = props => {
  const getContent = () => extractIconsOutOfChildren(props.children, Content, {
    isOutline: props.isOutline,
    isSoft: props.isSoft,
    palette: props.palette,
    shape: props.shape,
  })

  const [{ children, hasIconLeft, hasIconRight }, setContent] = React.useState(() => getContent())

  useChangeEffect(() => {
    setContent(getContent())
  }, [props.children, props.isSoft, props.palette, props.size, props.sizeSm, props.sizeMd, props.sizeLg, props.sizeXl])

  return (
    <BadgeElement
      badgeOffset={props.offset}
      badgePosition={props.position}
      badgeSize={props.size}
      badgeSizeSm={props.sizeSm}
      badgeSizeMd={props.sizeMd}
      badgeSizeLg={props.sizeLg}
      badgeSizeXl={props.sizeXl}
      hasIconLeft={hasIconLeft}
      hasIconRight={hasIconRight}
      hasSpace={props.hasSpace ?? !props.isAbsolute}
      isAbsolute={props.isAbsolute}
      isClickable={!!mouseClickEvents.find(i => props[i])}
      isFixedSize={props.isFixedSize}
      isOutline={props.isOutline}
      isSoft={props.isSoft}
      palette={props.palette}
      shape={props.shape}
      {...extractElementProps(props, [clickEvents, intrinsicStyledProps])}
    >
      {children}
    </BadgeElement>
  )
}

Badge.defaultProps = {
  offset: 10,
  palette: 'primary',
  position: Position.RIGHT,
  shape: 'rectangle',
  size: 'medium',
}

export default Badge
