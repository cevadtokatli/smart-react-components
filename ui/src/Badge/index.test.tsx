import { wrapTheme } from '@smart-react-components/core/test'
import { render } from '@testing-library/react'
import React from 'react'
import Badge from './'
import BadgeIcon from './BadgeIcon'

describe('<Badge />', () => {
  it('should render component', () => {
    const { asFragment } = render(wrapTheme(
      <Badge>
        <BadgeIcon><svg /></BadgeIcon>
        <span>Content</span>
        <BadgeIcon><svg /></BadgeIcon>
      </Badge>
    ))
    expect(asFragment()).toMatchSnapshot()
  })
})
