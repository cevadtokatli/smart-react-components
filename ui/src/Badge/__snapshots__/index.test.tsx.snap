// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render component 1`] = `
<DocumentFragment>
  .c0 {
  margin: .3125rem .3125rem;
}

.c0 .c4 {
  font-size: .75rem;
  padding: .1875rem .375rem;
  padding-left: 0.1875rem;
  padding-right: 0.1875rem;
}

.c0 .c2 {
  padding: 0 0.1875rem;
}

.c0 svg {
  height: .75rem;
  width: .75rem;
}

.c0 i {
  font-size: .75rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  border: solid 1px;
  border-right: 0;
  border-color: #0553AC;
  border-radius: .15rem;
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #0553AC;
  color: #fff;
  fill: #fff;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  border: solid 1px;
  border-left: 0;
  border-color: #0553AC;
  border-radius: .15rem;
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #0553AC;
  color: #fff;
  fill: #fff;
}

.c5 {
  border: solid 1px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  word-break: break-word;
  word-wrap: break-word;
}

.c1 .c4 {
  border-color: #0665d0;
  border-radius: .15rem;
}

.c1 .c4 {
  background: #0665d0;
  color: #fff;
  fill: #fff;
}

.c1 .c4 {
  border-left: 0;
  border-bottom-left-radius: 0;
  border-top-left-radius: 0;
  border-right: 0;
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

<div
    class="c0 c1"
    shape="rectangle"
  >
    <div
      class="c2  c3"
      shape="rectangle"
    >
      <svg />
    </div>
    <div
      class="c4 c5"
    >
      <span>
        Content
      </span>
    </div>
    <div
      class="c2  c6"
      shape="rectangle"
    >
      <svg />
    </div>
  </div>
</DocumentFragment>
`;
